### ISSUE TITLE
SSL on Defectdojo

### ISSUE THREAD

**Issue by vulnz** [DefectDojo/django-DefectDojo#5893]
Good day.
I try to access docker with nginx for Defectdojo and i clearly get permission denied on any file change.
Additionally I try to add nginx for defectdojo and because it is going to other nginx it looks like some kind of conflict.
My idea is super simple, i need to get SSL certificate to work in defecrdojo to enable saml.

Please let me know which steps do i need to take in order to add ssl cert to dojo.
Docker compose/docker file or other information which can help me to do it will be good, for example how can i access docker if it is denied. I don't want to regenerate docker 

https://github.com/DefectDojo/django-DefectDojo/blob/master/docker/entrypoint-nginx.sh

I have my own certs,  i'm ok to you any other certs for example selfsigned and i see here
https://github.com/DefectDojo/django-DefectDojo/blob/master/docker/entrypoint-nginx.sh it.

So basically how can i get defectodojo working with SSL and if there are any easy way

**Comment by vulnz** [comment]
if [ "${GENERATE_TLS_CERTIFICATE}" = true ]; then
  openssl req  \
      -x509 \
      -nodes \
      -days 365 \
      -newkey rsa:4096 \
      -keyout /etc/nginx/ssl/nginx.key \
      -out /etc/nginx/ssl/nginx.crt \
      -subj "/C=DE/ST=City/L=City/O=Global Security/OU=IT Department/CN=nginx"


how can i enable this or how can i add my cert there? by default defecto is on port 8080 without ssl


![image](https://user-images.githubusercontent.com/20043135/153179487-f6bf3d4c-9592-4535-b449-c2d24297f679.png)


This is what i see if i go on https: and port 8080

**Comment by vulnz** [comment]
I used docker.
How can i access https?

**Comment by valentijnscholten** [comment]
What happens if you follow the instructions at https://github.com/DefectDojo/django-DefectDojo/blob/master/readme-docs/DOCKER.md#run-with-docker-using-https

**Comment by El-Brujo-Manjaro** [comment]
Well. if you follow the instructions like they are right now, most likely, you'll get an error saying that the third-party certificates can't be found:
```
django-defectdojo-nginx-1 | 2022/03/24 20:28:42 [emerg] 1#1: cannot load certificate "/etc/nginx/ssl/nginx.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/ssl/nginx.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
django-defectdojo-nginx-1 | nginx: [emerg] cannot load certificate "/etc/nginx/ssl/nginx.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/ssl/nginx.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
```
The **solution** for me was:

1. Placing both certificate file and key file in a directory under the **nginx** folder, for example `nginx/ssl`
2. Adding a volume definition in the `docker-compose.override.https.yml` file so that it maps the container's nginx config path to the path above. For example:
```
volumes:
      - ./nginx/ssl:/etc/nginx/ssl
```
Hope the above helps to anybody else in the same situation.

**Comment by mtesauro** [comment]
FWIW, I prefer to bind-mount in the nginx config when doing docker-compose so you have full control of how nginx works - as show [here](https://github.com/DefectDojo/Community-Contribs/blob/master/CentOS-7-deploy/dojo/docker-compose.yml#L19).

Full example with commands, etc is at https://github.com/DefectDojo/Community-Contribs/tree/master/CentOS-7-deploy

**Comment by devGregA** [comment]
Closing — running DefectDojo behind HTTPS is covered by the deployment guidance shared above (reverse proxy / TLS termination). Please reopen or reach out on Slack if you need more help.
